vivisect/__init__.py

```diff
@@ -958,11 +958,16 @@ def parseOpcode(self, va, arch=envi.ARCH_DEFAULT):
     def iterJumpTable(self, startva):
```

Comment:
this is very specific.  have you considered making this size-settable (with the default of self.psize), and setting a maxcount arg?  that would allow for non-ptr-size JmpTables.  some compilers (amd64 in particular, probably MS but i'm brain-farting right now) iter through 32-bit offsets which are added/subtracted to some base.  this modification would make iterJmpTable() usable by even the symboliks-switchcase branch code.

Reply:
Good idea on the psize bit. I'm open to the maxcount arg, though I'm not entirely sure when it'll come in handy? All the test cases I've hit (especially since this iterJumpTable sits right in the middle of codeflow), we typically have no idea of what the size of the table is going to be. Adding it won't hurt, I'm just curious as to the use case.

Also, if you've got any samples of the nonptr size jump tables, I would love to add those to the test suite.